### DISCUSSION TITLE
Adding source to data submitted with logged listen?

### DISCUSSION THREAD

**Discussion by jbwharris** [foocorp/hacienda#102] [Future enhancements]
I'm curious if pairing the data source with a submitted song is possible? I have a rather idiosyncratic setup, but it would be simplified so much if I could split data by source.

1) I have 2 last.fm accounts, one for logging absolutely everything, and a second for logging everything I listen to outside of Apple Music. This can be from Web Scrobbler or Finale. The goal of splitting my data in this way is to make it so I can import the listening I do outside of Apple Music into my Apple Music library. If I didn't split the data, I'd be creating a recursive loop of reimporting songs I already have in my library. 
2) To do the syncing between the 2 accounts, this is where I use libre.fm. I log the plays to it, then sync those to my second last.fm account. For Spotify, since it allows multiple app integrations, I can log plays directly to the second account. 
3) I use SongShift to do the importing of my tracks to Apple Music. It has the ability to pull the 100 Recently Scrobbled, but this is a misnomer, as I'm pretty sure it'll submit up to 1000, the last.fm API limit. These songs go into my library and in turn immensely improve the recommendations I get from Apple Music. 

Now with all that said, I'd love to be able to identify the source that's logging the tracks, then be able to pull them out direct from libre.fm based on whether it was non-Apple Music plays.

**Comment by mattl** [comment]
This is an interesting setup. Ultimately I’d like to be able to offer something for this. Keeping this open.

**Comment by jbwharris** [comment]
Adding an addendum to this idea. Web Scrobbler would obviously be a source onto itself, but I'm almost wondering if it could get more granular. As you can see in the screenshot, a scrobble is labelled with which connector is being used, which would be cool info.

![image](https://github.com/user-attachments/assets/1688be0b-3f54-48f5-b2b0-7f897316c5d2)

As my site is [ScrobbleRadio](https://scrobblerad.io) and since it's built implicitly with a scrobbling first mentality, I'd love if I could pass a value for what radio station I'm listening to currently. I'm not sure if this becomes a public tag or categorization of the data, but imagine being able to search tags that would show scrobbles related to popular stations like KEXP, BBC6, WFUV, The Current etc. I think that would be a very valuable vein of info for a lot of people.
